<launch>
    <arg name="pathToScenario"/>
    <arg name="pathToOpenDrive" default="$(env HOME)/ros_scenario_simulation/src/sim_resources/map/MCubeTestMap.xodr"/>
    <arg name="pathForReport"/>
  

    <param name="pathForReport" value="$(arg pathForReport)"/>
    <param name="pathToScenario" value="$(arg pathToScenario)"/>
    <param name="pathToOpenDrive" value="$(arg pathToOpenDrive)"/>
    
    <!-- topic interface -->
    <include file="$(find scenario_evaluator)/launch/evaluator.launch">
        <arg name="pathToScenario" value="$(arg pathToScenario)"/>
        <arg name="pathForReport" value="$(arg pathForReport)"/>
    </include>

    <node pkg="scenario_sim" type="scenario_sim_node" name="sim_core" output="screen" required="true"/>

    <include file="$(find scenario_objects)/launch/scenario_objects.launch">
        <arg name="pathToScenario" value="$(arg pathToScenario)"/>
    </include>

    

    <include file="$(find scenario_goal)/launch/scenario_goal.launch"/>

    <node pkg="tf_broadcaster" type="tf_broadcaster_node" name="tf_broadcaster" />

    <!-- ROS-Nodes for freespace generation -->
    <include file="$(find path_to_freespace_converter)/launch/path_to_freespace_converter.launch">
    </include>
    <include file="$(find freespace_visualizer)/launch/freespace_visualizer.launch">
    </include>
    <include file="$(find freespace_goal)/launch/freespace_goal.launch">
    </include>


    <include file="$(find scenario_sim)/launch/ci_driving.launch">
        <arg name="opendriveFilePath" value="$(arg pathToOpenDrive)"/>
    </include>

</launch>